orchestrator/src/test/java/com/scottlogic/deg/orchestrator/cucumber/features/validation/CorrectConstraintTypes.feature

```diff
@@ -39,19 +34,11 @@ Then the profile is invalid because "Field \[foo\]: is type STRING , but you are
       | datetime| granularTo            | NUMERIC   | granular to 0.1                   | DATETIME  |
       | datetime| equalTo               | NUMERIC   | equal to 1                        | DATETIME    |
 
-      | datetime| matchingRegex         | STRING    | matching regex "ab"               | DATETIME  |
```

Comment:
I think we need to have a think if we should support regexs for datetime. Currently if you try with a profile like

```  {
    "schemaVersion": "0.9",
    "fields": [
      {
        "name": "date",
        "type": "datetime",
        "nullable": false
      }
    ],
    "rules": [
      {
        "constraints": [
          {
            "field": "date",
            "is": "matchingRegex",
            "value": "2000-01-01T00:00:00.000Z"
          }
        ]
      }
    ]
  }
```

then you get a java cast exception.

If you try with a profile containing a "real" regex like
```
  {
    "schemaVersion": "0.9",
    "fields": [
      {
        "name": "date",
        "type": "datetime",
        "nullable": false
      }
    ],
    "rules": [
      {
        "constraints": [
          {
            "field": "date",
            "is": "matchingRegex",
            "value": "200.-01-01T00:00:00.000Z"
          }
        ]
      }
    ]
  }
```

Then the profile fails validation.

Reply:
Actually we figured that that this is an issue in ConstraintValueValidator. Paul will raise an issue tomorrow.

However the regex line should prolly be readded